### PR TITLE
mfkey32: fix result inconsistency with cli

### PR DESCRIPTION
**Pull request by hank9999**
mfkey32: fix result inconsistency with cli
fix detection data incomplete getting
fix mf1GetDetectionResult command data wrong type (from Int16 to Int32)

### CODE REVIEW

File: chameleonultragui/lib/gui/page/mfkey32.dart [hunk 1]
```diff
@@ -62,8 +62,20 @@ class Mfkey32PageState extends State<Mfkey32Page> {
 
   Future<void> handleMfkeyCalculation() async {
     var appState = context.read<ChameleonGUIState>();
-
-    var detections = await appState.communicator!.getMf1DetectionResult(0);
+    var count = (await appState.communicator!.getMf1DetectionCount()).toInt();
```
Comment (Foxushka) [inline chameleonultragui/lib/gui/page/mfkey32.dart:65]: Isn't it already int?

File: chameleonultragui/lib/gui/page/mfkey32.dart [hunk 2]
```diff
@@ -62,8 +62,20 @@ class Mfkey32PageState extends State<Mfkey32Page> {
...
+    var rawDetectionResult = [];
+    var index = 0;
+    while (true) {
+      var tmpRawDetectionResult = await appState.communicator!
+          .getMf1DetectionResult(index);
+      index += tmpRawDetectionResult.length;
+      rawDetectionResult.addAll(tmpRawDetectionResult);
+      if (index >= count) {
```
Comment (Foxushka) [inline chameleonultragui/lib/gui/page/mfkey32.dart:73]: Migrate to `for (int index = 0; index < count; index++)`

File: chameleonultragui/lib/gui/page/mfkey32.dart [hunk 3]
```diff
@@ -62,8 +62,20 @@ class Mfkey32PageState extends State<Mfkey32Page> {
...
+        break;
+      }
+    }
+    var detections = (await appState.communicator!
+        .parseMf1DetectionResult(rawDetectionResult));
```
Comment (Foxushka) [inline chameleonultragui/lib/gui/page/mfkey32.dart:78]: No. Move all inside one function in communicator.

**Review by Foxushka — APPROVED** [review]
Completely don't like how you did it

**Review by Foxushka — CHANGES_REQUESTED** [review]
Oops, wrong button

**Comment by hank9999** [comment]
> Completely don't like how you did it

Sorry about that, I am new in Flutter and Dart. I will force update asap.

**Review by Foxushka — APPROVED** [review]
Thank you

**Pull request merged**